pkg/workload/core.go

```diff
@@ -265,6 +265,14 @@ func (c *core) DoInsert(ctx context.Context, db ycsb.DB) error {
 			break
 		}
 
+		select {
+		case <-ctx.Done():
+			if ctx.Err() == context.Canceled {
```

Comment:
seem we don't need to check Canceled here, if Done, we can return the ctx.Err directly

Reply:
Sorry if I didn't make it clear...
Timeout and Cancel both make ctx Done. But only Cancel should break the loop. Timeout should be handled by retrying via `insertionRetryLimit`. So here we need to make sure it's Cancel.